boofuzz/sessions.py

```diff
@@ -51,30 +52,58 @@ class Target(object):
 
     :param connection: Connection to system under test.
     :type connection: itarget_connection.ITargetConnection
+    :param monitors: List of Monitors for this Target.
+    :type monitors: List[Union[IMonitor, pedrpc.Client]]
+    :param monitor_alive: List of Functions that are called when a Monitor is alive. It is passed
+                          the monitor instance that became alive. Use it to e.g. set options
+                          on restart.
     :param repeater: Repeater to use for sending. Default None.
     :type repeater: repeater.Repeater
 
     """
 
-    def __init__(
-        self, connection, procmon=None, procmon_options=None, netmon=None, max_recv_bytes=10000, repeater=None
-    ):
+    def __init__(self, connection, monitors=[], monitor_alive=[], max_recv_bytes=10000, repeater=None, **kwargs):
```

Comment:
Default argument values are mutable.

Something like this would be better:
```python
def __init__(self, connection, monitors=None, monitor_alive=None, max_recv_bytes=10000, repeater=None, **kwargs):
    if monitors is None:
        monitors = []
    if monitor_alive is None:
        monitor_alive = []
```